stdlib/src/utils/static_tuple.mojo

```diff
@@ -240,3 +240,162 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
+            )
+
+    @always_inline("nodebug")
+    fn __len__(self) -> Int:
+        """Returns the length of the array. This is a known constant value.
+
+        Returns:
+            The size of the list.
+        """
+        return size
+
+    @always_inline("nodebug")
+    fn _get_reference_unsafe[
+        mutability: __mlir_type.i1,
+        self_life: AnyLifetime[mutability].type,
+    ](
+        self: Reference[Self, mutability, self_life]._mlir_type, index: Int
+    ) -> Reference[Self.ElementType, mutability, self_life]:
+        """Get a reference to an element of self without checking index bounds.
+
+        Users should opt for `__refitem__` instead of this method.
+        """
+        var ptr = __mlir_op.`pop.array.gep`(
+            Reference(Reference(self)[]._array).get_legacy_pointer().address,
+            index.value,
+        )
+        return Reference[Self.ElementType, mutability, self_life](
+            UnsafePointer(ptr)[]
+        )
+
+    @always_inline("nodebug")
+    fn __refitem__[
+        mutability: __mlir_type.i1,
+        self_life: AnyLifetime[mutability].type,
+        IntableType: Intable,
+    ](
+        self: Reference[Self, mutability, self_life]._mlir_type,
+        index: IntableType,
+    ) -> Reference[Self.ElementType, mutability, self_life]:
+        """Get a `Reference` to the element at the given index.
+
+        Parameters:
+            mutability: The inferred mutability of the reference.
+            self_life: The inferred lifetime of the reference.
+            IntableType: The inferred type of an intable argument.
+
+        Args:
+            index: The index of the item.
+
+        Returns:
+            A reference to the item at the given index.
+        """
+        debug_assert(-size <= int(index) < size, "Index must be within bounds.")
+        var normalized_idx = int(index)
+        if normalized_idx < 0:
+            normalized_idx += size
+
+        return Reference(self)[]._get_reference_unsafe[mutability, self_life](
+            normalized_idx
+        )
+
+    @always_inline("nodebug")
+    fn __refitem__[
+        mutability: __mlir_type.i1,
+        self_life: AnyLifetime[mutability].type,
+        IntableType: Intable,
+        index: IntableType,
+    ](self: Reference[Self, mutability, self_life]._mlir_type) -> Reference[
+        Self.ElementType, mutability, self_life
+    ]:
+        """Get a `Reference` to the element at the given index.
+
+        Parameters:
+            mutability: The inferred mutability of the reference.
+            self_life: The inferred lifetime of the reference.
+            IntableType: The inferred type of an intable argument.
+            index: The index of the item.
+
+        Returns:
+            A reference to the item at the given index.
+        """
+        alias i = int(index)
+        constrained[-size <= i < size, "Index must be within bounds."]()
+
+        var normalized_idx = i
```

Comment:
**Quibble** I think we could do this arithmetic at compile time since `size` is a compile time parameter.  But totally fine for now. 😄